network/src/commonJvmAndroid/kotlin/com/wire/kalium/network/HttpEngine.kt

```diff
@@ -0,0 +1,35 @@
+package com.wire.kalium.network
+
+import io.ktor.client.engine.HttpClientEngine
+import io.ktor.client.engine.okhttp.OkHttp
+import okhttp3.Interceptor
+import okhttp3.Response
+
+actual fun defaultHttpEngine(): HttpClientEngine {
+    return OkHttp.create {
+        /**
+         * This is a hack
+         *
+         * As seen here: https://github.com/ktorio/ktor/issues/1127
+         * Ktor is pretty has `application/protobuf` but not `application/x-protobuf`
+         * and it doesn't support custom content types!
+         */
+        addInterceptor(object : Interceptor {
+            override fun intercept(chain: Interceptor.Chain): Response {
+                val currentRequest = chain.call().request()
+                val currentHeaders = chain.call().request().headers
+                // When sending a byteArray, Ktor will put `octet-stream`. We can identify this and replace with protobuf
+                val isProtoBuf = currentHeaders["Content-Type"] == "application/octet-stream"
```

Comment:
what will happen when do need to send `application/octet-stream` hint upload asset will BE complain about having an `application/x-protobuf` in the headers in that case?

Reply:
The same that would happens when we want to send `application/x-protobuf` and Ktor+OkHttp sends an `application/octet-stream`: The server returns a `404`, saying it can't find an endpoint (that takes this content).

There are currently no endpoints that take an `application/octet-stream`. So this is a problem I would leave for the older and more experienced versions of ourselves to solve. Hopefully Ktor will have a better support for this in the future.

Reply:
[post:provider/assets](https://staging-nginz-https.zinfra.io/api/swagger-ui/#/default/post_provider_assets)
> A body with content type multipart/mixed body. The first section's content type should be application/json. The second section's content type should be always be application/octet-stream. Other content types will be ignored by the server.

Reply:
ok i did try it and the upload was successful, so maybe BE does not care 
anyway i would keep an eye for this one

Reply:
Oooooooh I should be more careful. I only saw the main request (`multipart/mixed`).

Indeed!

I've tested it manually here, and because [we just put stuff into the body](https://github.com/wireapp/kalium/blob/70e8d4f499f1ba3084dbde832880dea037cec064/network/src/commonMain/kotlin/com/wire/kalium/network/api/asset/AssetApiImpl.kt#L54), this interceptor doesn't interfere.

So everything works 😎